### PR TITLE
Get front end tests passing

### PR DESCRIPTION
**Pull request by tonyaellie**
## What type of PR is this?

_(REQUIRED)_

- cleanup

## What this PR does / why we need it:

_(REQUIRED)_

- Mostly sorts issues with linting and fixes one of the tests.

## Release Notes

- **New Features**
	- Enhanced styling for markdown elements, including headings and lists.
	- Improved scrollbar appearance with custom styles.
	- Dynamic HTML content added to the tools page for better user guidance.
	- New computed properties for better management of item details and user interactions.

- **Bug Fixes**
	- Streamlined error handling during item creation and updates to prevent multiple submissions.
	- Adjusted conditions for displaying character counts in text fields and areas.
	- Improved handling of asynchronous data fetching for item retrieval.

- **Documentation**
	- Improved code readability and structure across various components, enhancing maintainability.

- **Style**
	- Reordered class attributes for better CSS specificity and rendering order across multiple components.
	- Reformatted components for improved readability, including `FormTextArea`, `FormTextField`, and various modals.

- **Chores**
	- Cleaned up unused styles and improved formatting in several files for consistency.
	- Added a new dependency to enhance development capabilities.
	- Updated type checking script in package configuration.

### CODE REVIEW

**Comment by tonyaellie** [comment]
Still need to test none of these changes have caused any bugs.

File: frontend/components/Base/SectionHeader.vue [hunk 1]
```diff
@@ -4,7 +4,6 @@
       class="flex items-center text-3xl font-bold tracking-tight"
       :class="{
         'text-neutral-content': dark,
-        'text-content': !dark,
```
Comment (tonyaellie) [inline frontend/components/Base/SectionHeader.vue:7]: This doesn't seem to do anything so removed it.

File: frontend/components/Form/Autocomplete2.vue [hunk 2]
```diff
@@ -170,9 +170,11 @@
      * Resolve the issue of language not being supported
      */
     for (let i = 0; i < props.items.length; i++) {
-      const item = props.items[i]
-      if(resultIDs.find(item_ => item_ === item.id) != undefined){continue}
-      if(item.treeString.indexOf(search.value) > -1){
+      const item = props.items[i];
+      if (resultIDs.find(item_ => item_ === item.id) !== undefined) {
```
Comment (tonyaellie) [inline frontend/components/Form/Autocomplete2.vue:177]: Need to check that using a strict compare doesn't break it.

File: frontend/components/Form/TextArea.vue [hunk 3]
```diff
@@ -63,10 +63,12 @@
     },
     maxLength: {
       type: Number,
+      default: -1,
```
Comment (tonyaellie) [inline frontend/components/Form/TextArea.vue:66]: Not sure if -1 is the best default.

File: frontend/lib/api/__test__/user/items.test.ts [hunk 4]
```diff
@@ -148,9 +148,7 @@ describe("user should be able to create an item and add an attachment", () => {
     {
       const { response, data } = await api.items.maintenance.getLog(item.id);
       expect(response.status).toBe(200);
-      expect(data.entries).toHaveLength(maintenanceEntries.length);
-      expect(data.costAverage).toBeGreaterThan(0);
-      expect(data.costTotal).toBeGreaterThan(0);
+      expect(data).toHaveLength(maintenanceEntries.length);
```
Comment (tonyaellie) [inline frontend/lib/api/__test__/user/items.test.ts:151]: Data seems to be an array, maybe this was previosly an object?

File: frontend/pages/tools.vue [hunk 5]
```diff
@@ -39,6 +39,7 @@
         <div class="divide-y divide-gray-300 border-t border-gray-300 px-6 pb-3">
           <DetailAction @action="modals.import = true">
             <template #title> {{ $t("tools.import_export_set.import") }} </template>
+            <!-- eslint-disable-next-line vue/no-v-html -->
```
Comment (tonyaellie) [inline frontend/pages/tools.vue:42]: Not sure whether we should switch to using templates see (https://stackoverflow.com/questions/70751297/i18n-translate-a-sentence-that-one-of-words-is-bold).

File: frontend/components/Form/TextArea.vue (hunk 3 above)
Comment (tankerkiller125) [reply frontend/components/Form/TextArea.vue:66]: I'm not entirely sure on this one either, either `-1` or maybe `Number.MAX_VALUE` from Typescript?

File: frontend/pages/tools.vue (hunk 5 above)
Comment (tankerkiller125) [reply frontend/pages/tools.vue:42]: I didn't see the template thing in the documentation I was reading https://i18n.nuxtjs.org/docs/getting-started however, if it works, I think it would make sense to switch.

File: frontend/components/Form/TextArea.vue (hunk 3 above)
Comment (tonyaellie) [reply frontend/components/Form/TextArea.vue:66]: @katosdev thoughts?

File: frontend/pages/tools.vue (hunk 5 above)
Comment (tonyaellie) [reply frontend/pages/tools.vue:42]: Can't seem to get this to work so I'll leave it for now, not too big of an issue anyway.

**Review by tankerkiller125 — APPROVED** [review]
